test/function/samples/external-ignore-reserved-null-marker/_config.js

```diff
@@ -0,0 +1,25 @@
+module.exports = {
+	description: 'external function ignores \\0 started ids',
+	options: {
+		external(id) {
+			if (id.startsWith('\0')) {
+				throw Error('\\0 started ids should not be tested as external');
+			}
+			return true;
+		},
+		plugins: [
+			{
+				resolveId(importee) {
+					if (importee === '\0external') {
+						return importee;
+					}
+				},
```

Comment:
As I think for this change, the test is very important as documentation, maybe you could change it to better showcase when this typically happens? Not everyone is aware why this "\0" convention even exists so maybe instead of actually having a "\0" in the source .js files of the test, the "\0" could be generated by the same plugin that resolves it?